model/project_vars.go

```diff
@@ -280,7 +324,28 @@ func GetAWSKeyForProject(projectId string) (*AWSSSHKey, error) {
 	}, nil
 }
 
+// defaultParameterStoreAccessTimeout is the default timeout for accessing
+// Parameter Store. In general, the context timeout should prefer to be
+// inherited from a higher-level context (e.g. a REST request's context), so
+// this timeout should only be used as a last resort if the context cannot
+// easily be passed down.
+const defaultParameterStoreAccessTimeout = 30 * time.Second
+
 func (projectVars *ProjectVars) Upsert() (*adb.ChangeInfo, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultParameterStoreAccessTimeout)
+	defer cancel()
+	isPSEnabled, err := isParameterStoreEnabledForProject(ctx, projectVars.Id)
+	grip.Error(message.WrapError(err, message.Fields{
+		"message":    "could not check if Parameter Store was enabled for project; assuming it's disabled and falling back to using the DB",
+		"project_id": projectVars.Id,
+	}))
+	if isPSEnabled {
```

Comment:
Should we add a TODO here? Does that mean every time now we upsert project vars we do a log? Do you know if this would get noisy/fill up splunk? I don't think it would but something to consider

Reply:
Sounds good!